bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/bubble/Bubble.java

```diff
@@ -1,5 +1,229 @@
+/*
+ * File: Bubble.java
+ *
+ * Class: Bubble
+ *
+ * Version: 0.0.1
+ *
+ * Date: September 7th, 2016
+ *
+ */
+
+
 package nl.tudelft.bust_a_move20162017.bust_a_move_framework.bubble;
 
+import org.newdawn.slick.Color;
+import org.newdawn.slick.Graphics;
+
+/**
+ * The Bubble class represents a single bubble entity
+ *
+ * @author Calvin Nhieu
+ *
+ */
 public class Bubble {
+  public static final double DIAMETER = 40;
+  public static final double SPEED = 5;
+  private static final Color RED_COLOR = Color.red;
+  private static final Color BLUE_COLOR = Color.blue;
+  private static final Color YELLOW_COLOR = Color.yellow;
+  private static final Color GREEN_COLOR = Color.green;
+  private double x;
+  private double y;
+  private double xSpeed;
+  private double ySpeed;
+  private int angle;
+  private ColorChoice color;
+  private Color drawColor;
+  private State state;
+
+  public enum State {
+    STILL, FIRING, POPPING, DROPPING
+  }
+  public enum ColorChoice {
+    RED, BLUE, YELLOW, GREEN
+  }
+
+  /**
+   * Creates Bubble instance
+   *
+   * @param x  double value for starting x position
+   * @param y  double value for starting y position
+   * @param color  ColorChoice enum value for bubble's color
+   */
+  public Bubble(double x, double y, ColorChoice color) {
+    this.x = x;
+    this.y = y;
+    this.xSpeed = 0;
+    this.ySpeed = 0;
+    this.angle = 0;
+    this.color = color;
+    this.state = State.STILL;
+
+    switch(color) {
+      case RED:
+        this.drawColor = Bubble.RED_COLOR;
+        break;
+      case BLUE:
+        this.drawColor = Bubble.BLUE_COLOR;
+        break;
+      case YELLOW:
+        this.drawColor = Bubble.YELLOW_COLOR;
+        break;
+      case GREEN:
+        this.drawColor = Bubble.GREEN_COLOR;
+        break;
+    }
+  }
+
+  /**
+   * draws the Bubble
+   *
+   * @param g  Java Graphics instance
+   */
+  public void draw(Graphics g) {
+    g.setColor(this.drawColor);
+    g.fillOval((int) this.x, (int) this.y, (int) Bubble.DIAMETER, (int) Bubble.DIAMETER);
```

Comment:
Make it: 		g.fillOval((int) (this.x-Bubble.DIAMETER/2), (int) (this.y-Bubble.DIAMETER/2)
				, (int) Bubble.DIAMETER, (int) Bubble.DIAMETER); cause the x and y is the left top coördinate